src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -315,22 +298,47 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
   }
 
   @Override
-  public ResponseEntity<JobModel> createGoogleContext(
-      UUID id, @Valid CreateGoogleContextRequestBody body) {
+  public ResponseEntity<JobReport> createCloudContext(
+      UUID id, @Valid CreateCloudContextRequest body) {
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    ControllerValidationUtils.validateCloudContext(body.getCloudContext());
+    String jobId = body.getJobControl().getId();
+    String resultRelativePath = getAsyncResultEndpoint(jobId);
+
+    workspaceService.createGoogleContext(id, jobId, resultRelativePath, userReq);
+    JobReport jobReport = jobService.retrieveJob(jobId, userReq);
+    return ResponseEntity.status(HttpStatus.ACCEPTED).body(jobReport);
+  }
+
+  @Override
+  public ResponseEntity<CreateCloudContextResult> createCloudContextResult(UUID id, String jobId) {
     AuthenticatedUserRequest userReq = getAuthenticatedInfo();
-    // TODO(PF-153): Use the optional jobId from the body for idempotency instead of always creating
-    // a new job id.
-    String jobId = workspaceService.createGoogleContext(id, userReq);
-    JobModel jobModel = jobService.retrieveJob(jobId, userReq);
-    // TODO(PF-221): Fix the jobs polling location once it exists.
-    return ResponseEntity.status(HttpStatus.ACCEPTED)
-        .location(URI.create(String.format("/api/jobs/v1/%s", jobId)))
-        .body(jobModel);
+    JobReport jobReport = jobService.retrieveJob(jobId, userReq);
+    CreateCloudContextResult response = new CreateCloudContextResult().jobReport(jobReport);
+    // If the job is still running, return a 202 with just the JobReport.
+    if (jobReport.getStatus().equals(StatusEnum.RUNNING)) {
+      return new ResponseEntity<>(response, HttpStatus.ACCEPTED);
+    }
+    // This flight does not directly output a result, but may still throw errors. Either the
+    // errorReport or the googleContext field will be populated in the response, but never both.
+    JobResultOrException resultOrException = jobService.retrieveJobResult(jobId, null, userReq);
+    if (resultOrException.getException() != null) {
```

Comment:
One issue with that approach is that every caller to JobService (including WM's other synchronous flights) would then need to implement the same logic of checking JobReport status and calling the right endpoint. If you move that logic into JobService, I don't know what the return type could be besides JobResultOrException.

I re-did this section a bit to move more generic job code into JobService, but there's still more `null` usage than I'd really like, and this might need more thought.